libdnf/module/PlatformModulePackage.cpp

```diff
@@ -0,0 +1,70 @@
+#include "PlatformModulePackage.hpp"
+#include "libdnf/utils/File.hpp"
+
+#include <algorithm>
+#include <sstream>
+
+Id PlatformModulePackage::createSolvable(Pool *pool, HyRepo repo, const std::string &osReleasePath,
+                                         const std::string &arch)
+{
+    Id id = repo_add_solvable(repo->libsolv_repo);
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    std::string name = "platform";
+    std::string stream = getPlatformStream(osReleasePath);
+    long version = 1;
+    std::string context = "0";
+
+    std::ostringstream ss;
+    ss << name << "-" << stream << "-" << version << "-" << context;
+    solvable_set_str(solvable, SOLVABLE_NAME, ss.str().c_str());
+    solvable_set_str(solvable, SOLVABLE_EVR, "0");
+    solvable_set_str(solvable, SOLVABLE_ARCH, arch.c_str());
+
+    ss.str(std::string());
+    ss << "module-n(" << name << ")";
```

Comment:
please don't use this form. use `module(n)`, `module(n:s)` and such

Reply:
Can you provide any explanation why your suggestion is better and which benefits it brings over existing code in the PR?

Reply:
**consistency**

And also smaller memory footprint. E.g. because libsolv optimizes stringpool.